src/include/bootloader/routines/misc.rb

```diff
@@ -227,32 +220,6 @@ def restoreMBR(device)
       end
     end
 
-    # Get map of swap partitions
-    # @return a map where key is partition name and value its size
-    def getSwapPartitions
-      # FIXME: move to boot storage
-      tm = Storage.GetTargetMap
-      ret = {}
-      tm.each_value do |v|
-        cyl_size = v["cyl_size"] || 0
-        partitions = v["partitions"] || []
-        partitions = partitions.select do |p|
-          p["mount"] == "swap" && !p["delete"]
-        end
-        partitions.each do |s|
-          # bnc#577127 - Encrypted swap is not properly set up as resume device
-          if s["crypt_device"] && !s["crypt_device"].empty?
-            dev = s["crypt_device"]
-          else
-            dev = s["device"]
-          end
-          ret[dev] = Ops.get_integer(s, ["region", 1], 0) * cyl_size
```

Comment:
I guess this code was there before partition["size_k"] was introduced in the target map. Since there is no comment stating the contrary, I'll assume it can be safely replaced by v["size_k"].